src/content/1.7/modules/sample_modules/view-order-hooks-demo.md

```diff
@@ -0,0 +1,522 @@
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Install;
+
+use Db;
+use Module;
+
+/**
+ * Class responsible for modifications needed during installation/uninstallation of the module.
+ */
+class Installer
+{
+    /**
+     * @var FixturesInstaller
+     */
+    private $fixturesInstaller;
+
+    public function __construct(FixturesInstaller $fixturesInstaller)
+    {
+        $this->fixturesInstaller = $fixturesInstaller;
+    }
+
+    /**
+     * Module's installation entry point.
+     *
+     * @param Module $module
+     *
+     * @return bool
+     */
+    public function install(Module $module): bool
+    {
+        if (!$this->registerHooks($module)) {
+            return false;
+        }
+
+        if (!$this->installDatabase()) {
+            return false;
+        }
+
+        $this->fixturesInstaller->install();
+
+        return true;
+    }
+
+    /**
+     * A helper that executes multiple database queries.
+     *
+     * @param array $queries
+     *
+     * @return bool
+     */
+    private function executeQueries(array $queries): bool
+    {
+        foreach ($queries as $query) {
+            if (!Db::getInstance()->execute($query)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Install the database modifications required for this module.
+     *
+     * @return bool
+     */
+    private function installDatabase(): bool
+    {
+        $queries = [
+            'CREATE TABLE IF NOT EXISTS `'._DB_PREFIX_.'signature` (
+              `id_signature` int(11) NOT NULL AUTO_INCREMENT,
+              `id_order` int(11) NOT NULL,
+              `filename` varchar(64) NOT NULL,
+              PRIMARY KEY (`id_signature`),
+              UNIQUE KEY (`id_order`)
+            ) ENGINE='._MYSQL_ENGINE_.' DEFAULT CHARSET=utf8;'
+        return $this->executeQueries($queries);
+    }
+}
+```
+Also let's create `FixturesInstaller` class for populating our database with data
+
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Install;
+
+use Db;
+use Order;
+
+/**
+ * Installs data fixtures for the module.
+ */
+class FixturesInstaller
+{
+    /**
+     * @var Db
+     */
+    private $db;
+
+    public function __construct(Db $db)
+    {
+        $this->db = $db;
+    }
+
+    public function install(): void
+    {
+        $orderIds = Order::getOrdersIdByDate('2000-01-01', '2100-01-01');
+
+        foreach ($orderIds as $orderId) {
+            $this->insertSignature($orderId);
+        }
+    }
+
+    private function insertSignature(int $orderId): void
+    {
+        $this->db->insert('signature', [
+            'id_order' => $orderId,
+            'filename' => 'john_doe.png',
+        ]);
+    }
+}
+
+```
+Then let's create `InstallerFactory` which will be used both to install and uninstall database:
```

Comment:
Its not used directly to install or unninstal, factory is used to create the installer